### PR TITLE
add color_quant

### PR DESCRIPTION
**Pull request by sammyne**
### Description

Add color_quant@master

### Crate Meta

- **Name**: color_quant
- **Version**: master
- **License**: MIT
- **Upstream commit hash base**: 3a1d0c129301adcc1ad4caa91e19089294721c58
- **Teaclave SGX SDK version**: v1.1.1

## NOTICES
The original project failed to compile when upgrading to rust 2018, so this commit consists of two commits

1. format and fix syntax errors for upgrading
2. port to teaclave-sgx-sdk

Hence, the Diff of Changes reported below is the difference of 2 from 1.

### Diff of Changes

```diff
diff --git a/crates/color_quant/Cargo.toml b/crates/color_quant/Cargo.toml
index d1e95ab..13502e6 100644
--- a/crates/color_quant/Cargo.toml
+++ b/crates/color_quant/Cargo.toml
@@ -6,4 +6,13 @@ authors = ["nwin <nwin@users.noreply.github.com>"]
 readme = "README.md"
 description = "Color quantization library to reduce n colors to 256 colors."
 repository = "https://github.com/PistonDevelopers/color_quant.git"
-edition = "2018"
\ No newline at end of file
+edition = "2018"
+
+[features]
+default = ["mesalock_sgx"]
+mesalock_sgx = ["sgx_tstd"]
+enclave_unit_test = ["sgx_tunittest"]
+
+[dependencies]
+sgx_tstd        = { git = "https://github.com/apache/incubator-teaclave-sgx-sdk.git", tag = "v1.1.1", optional = true }
+sgx_tunittest   = { git = "https://github.com/apache/incubator-teaclave-sgx-sdk.git", tag = "v1.1.1", optional = true }
diff --git a/crates/color_quant/src/lib.rs b/crates/color_quant/src/lib.rs
index fbc6b4b..593c2bc 100644
--- a/crates/color_quant/src/lib.rs
+++ b/crates/color_quant/src/lib.rs
@@ -60,6 +60,17 @@ that this copyright notice remain intact.
 //! let color_map = nq.color_map_rgba();
 //! ```
 //!
+
+#![cfg_attr(all(feature = "mesalock_sgx", not(target_env = "sgx")), no_std)]
+#![cfg_attr(
+    all(target_env = "sgx", target_vendor = "mesalock"),
+    feature(rustc_private)
+)]
+#[cfg(all(feature = "mesalock_sgx", not(target_env = "sgx")))]
+#[macro_use]
+extern crate sgx_tstd as std;
+use std::prelude::v1::*;
+
 use std::cmp::{max, min};
 
 macro_rules! clamp(
```

### CODE REVIEW

File: meta.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ project name, version, upstream URL, upstream commit hash base, license, tests (
 adler32-rs, 1.0.4, https://github.com/remram44/adler32-rs, b235700967618065f89f188a0000d67d08451705, zlib, full
 anyhow, 1.0.28,https://github.com/dtolnay/anyhow, 022f700a1c67c5cd663c34b8a3c11c6818ec0ecb, MIT/Apache-2.0, partial
 cfg-if, 0.1.10, https://github.com/alexcrichton/cfg-if, 4484a6faf816ff8058088ad857b0c6bb2f4b02b2, MIT/Apache-2.0, full
+color_quant, master, 3a1d0c129301adcc1ad4caa91e19089294721c58, MIT, no
```
Comment (mssun) [inline meta.txt]: upstream URL? test should be `none`.

Comment (sammyne) [reply meta.txt]: Added now~ @mssun

Comment (mssun) [reply meta.txt]: Test should be one of `none, partial, full, none-no-std`. This is `none`, but not `no`.

Comment (sammyne) [reply meta.txt]: Updated~

**Pull request merged**